rebalance.go

```diff
@@ -0,0 +1,173 @@
+	Status    RebalStatus `msg:"status"`  // Current state of rebalance operation. One of Started|Stopped|Completed|Failed.
+}
+
+type RebalanceStats struct {
+	InitFreeSpace uint64 `json:"initFreeSpace"` // Pool free space at the start of rebalance
+	InitCapacity  uint64 `json:"initCapacity"`  // Pool capacity at the start of rebalance
+
+	Buckets           []string      `json:"buckets"`           // buckets being rebalanced or to be rebalanced
+	RebalancedBuckets []string      `json:"rebalancedBuckets"` // buckets rebalanced
+	Bucket            string        `json:"bucket"`            // Last rebalanced bucket
+	Object            string        `json:"object"`            // Last rebalanced object
+	NumObjects        uint64        `json:"numObjects"`        // Number of objects rebalanced
+	NumVersions       uint64        `json:"numVersions"`       // Number of versions rebalanced
+	Bytes             uint64        `json:"bytes"`             // Number of bytes rebalanced
+	Participating     bool          `json:"participating"`
+	Info              RebalanceInfo `json:"info"`
+}
+
+type RebalanceMeta struct {
+	ARN             [16]byte          `json:"arn"`             // ARN of the ongoing rebalance operation
+	PercentFreeGoal float64           `json:"percentFreeGoal"` // Computed from total free space and capacity at the start of rebalance
+	PoolStats       []*RebalanceStats `json:"poolStats"`       // Per-pool rebalance stats keyed by pool index
+	StoppedAt       time.Time         `json:"stoppedAt"`
+}
+
+// rebalMeta used to specialize marshaling/unmarshaling of uuid.
+type rebalMeta struct {
+	ARN             uuid.UUID         `json:"arn"`
+	PercentFreeGoal float64           `json:"percentFreeGoal"`
+	PoolStats       []*RebalanceStats `json:"poolStats"`
+	StoppedAt       time.Time         `json:"stoppedAt"`
+}
+
+func (rm *RebalanceMeta) MarshalJSON() ([]byte, error) {
+	r := rebalMeta{
+		ARN:             uuid.UUID(rm.ARN),
+		PercentFreeGoal: rm.PercentFreeGoal,
+		PoolStats:       rm.PoolStats,
+		StoppedAt:       rm.StoppedAt,
+	}
+	return json.Marshal(r)
+}
+
+func (rm *RebalanceMeta) UnmarshalJSON(b []byte) error {
+	r := rebalMeta{}
+	err := json.Unmarshal(b, &r)
+	if err != nil {
+		return err
+	}
+
+	copy(rm.ARN[:], r.ARN[:])
+	rm.PercentFreeGoal = r.PercentFreeGoal
+	rm.PoolStats = r.PoolStats
+	rm.StoppedAt = r.StoppedAt
+	return nil
+}
+
+func (adm *AdminClient) RebalanceStart(ctx context.Context) (arn uuid.UUID, err error) {
+	// Execute POST on /minio/admin/v3/rebalance/start to start a rebalance operation.
+	var resp *http.Response
+	resp, err = adm.executeMethod(ctx,
+		http.MethodPost,
+		requestData{relPath: adminAPIPrefix + "/rebalance/start"})
+	defer closeResponse(resp)
+	if err != nil {
+		return arn, err
+	}
+
+	if resp.StatusCode != http.StatusOK {
+		return arn, httpRespToErrorResponse(resp)
+	}
+
+	var rebalInfo struct {
+		ARN uuid.UUID `json:"arn"`
```

Comment:
ARN has a specific meaning in AWS S3 spec not sure it would be wise to use UUID as ARN - you can just call it as ID @krisis